codeclimate-wrapper

```diff
@@ -41,17 +41,34 @@ else
     && invalid_docker_host "$DOCKER_HOST"
 fi
 
+CC_XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-~/.config/}
+CC_XDG_CONFIG="${CC_XDG_CONFIG_HOME}/codeclimate/config.yml"
+CC_XDG_CONFIG_DIR=$(dirname "${CC_XDG_CONFIG}")
+[ -d "$CC_XDG_CONFIG_DIR" ] || mkdir -p "$CC_XDG_CONFIG_DIR"
+[ -f "$CC_XDG_CONFIG" ] || ( echo "---" > "$CC_XDG_CONFIG" )
+
+CC_XDG_CACHE_HOME=${XDG_CACHE_HOME:-~/.cache/}
+CC_XDG_CACHE="${CC_XDG_CACHE_HOME}/codeclimate/cache.yml"
+CC_XDG_CACHE_DIR=$(dirname "${CC_XDG_CACHE}")
+[ -d "$CC_XDG_CACHE_DIR" ] || mkdir -p "$CC_XDG_CACHE_DIR"
+[ -f "$CC_XDG_CACHE" ] || ( echo "---" > "$CC_XDG_CACHE" )
+
 docker_run() {
-  exec docker run \
+  docker run \
```

Comment:
What this intentional? It drastically changes the process tree semantics, signal handling, etc.